JotunnLib/Utils/ModCompatibility/ModCompatibility.cs

```diff
@@ -30,6 +32,34 @@ internal static void Init()
             Main.Harmony.PatchAll(typeof(ModCompatibility));
         }
 
+        public static bool IsJotunnOnServer()
+        {
+            return IsModuleOnServer(Main.ModGuid);
+        }
+
+
+        public static bool IsModuleOnServer(BaseUnityPlugin plugin)
+        {
+            return IsModuleOnServer(plugin.Info.Metadata.GUID);
+        }
+
+        public static bool IsModuleOnServer(string modGUID)
+        {
+            if (ZNet.instance && ZNet.instance.IsClientInstance())
+            {
+                if (LastServerVersion == null)
+                {
+                    return false;
+                }
+                // maybe the HashSet of Module GUIDs should be cached along with LastServerVersion
+                // and reset along with it to, just to avoid construction of the HashSet
+                var serverData = new ModuleVersionData(LastServerVersion);
+                var moduleGUIDs = new HashSet<string>(serverData.Modules.Select(mod => mod.guid).ToList());
+                return moduleGUIDs.Contains(modGUID);
```

Comment:
`IsModuleOnServer` is definitively called way too often to include such an expensive operation. This has to be cached, probably by always deserializing LastServerVersion instead of keeping the ZPackage

Reply:
I've just pushed some changes that implement and use a `ServerVersionData` class to hold the deserialized data and a cached copy of the GUID HashSet. I realize I may have to rollback the changes depending on how the RPC stuff pans out though as the `ServerVersionData` class relies on the new data layout of `ModModule` to get the GUIDs.